activerecord/lib/active_record/migration.rb

```diff
@@ -1395,11 +1402,14 @@ def validate(migrations)
       end
 
       def record_version_state_after_migrating(version)
-        if down?
+        if ENV["DRY_RUN_MIGRATION"] = "true"
+          # do nothing
+        elsif down?
```

Comment:
This is here to skip recording the version state after a dry-run migration has happened successfully. The state of the schema hasn’t changed, so we don’t want to change the version that implies that it did change.